library/common/main_interface.h

```diff
@@ -200,6 +200,8 @@ void terminate_engine(envoy_engine_t engine);
  */
 envoy_status_t drain_connections(envoy_engine_t engine);
 
+Envoy::Upstream::ClusterManager& get_cluster_manager(envoy_engine_t engine);
```

Comment:
This is the C interface, we can't return C++ types through it. I'd just remove this, callers will have to get it via the Engine handle directly

Reply:
and how do we access the engine object ? by calling this - https://github.com/envoyproxy/envoy-mobile/blob/e41c2b65c7d1c523b99675174a60c8f77393841a/library/common/main_interface.cc#L19 
Again, not sure if we can call it.

Reply:
Huh I was under the impression that this was accessible, seems like not. Some options:
- declare `engine()` as an exported function in a .h so that other code can pull out the internal engine handle
- similar to above, but return a cc/Engine that wraps the engine(). This would retain a bit more separation between internal and external APIs. We'd essentially let C++ get a handle to a C++ engine that points to the same underlying engine that was instantiated by the platform initialization call.
- have the C function return a void* which can be cast to ClusterManager* by calling code. This should work fine but is a bit gross in terms of APIs
- Rethink this in a way where we're not directly exposing the CM via the engine

Looping in @goaway for thoughts here

The context here is having external code integrate with the EM cluster manager, specifically in order to set up xDS subscriptions using the built in xDS subscription subsystem.

Reply:
I think allowing the engine to be exposed is fine for now, but I don't feel strongly about it.